### PR TITLE
Migrate to React 15.5

### LINKED ISSUE #29: Use 'create-react-class' package instead of React.createClass
`React.createClass` has been deprecated and extracted to it's own package since [React v15.5.0](https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html#new-deprecation-warnings).

The following warning is displayed if that version of React is used:
`Warning: Highlight: React.createClass is deprecated and will be removed in version 16. Use plain JavaScript classes instead. If you're not yet ready to migrate, create-react-class is available on npm as a drop-in replacement.`

See [Migrating from `React.createClass`](https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html#migrating-from-react.createclass) to fix this.

This could be a temporary fix, but I think moving to ES6 classes is more appropriate :-)

### LINKED ISSUE #30: Use 'react-dom/test-utils' package instead of 'react-addons-test-utils'
`react-addons-test-utils` has been deprecated and has been put in the `react-dom` package since [React v15.5.0](https://facebook.github.io/react/blog/#react-test-utils).

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Closes #29 & closes #30

CC: @akiran

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
@akiran I think this can be merged isn't it?

File: src/index.js [hunk 1]
```diff
@@ -1,51 +1,50 @@
...
+            } else {
+                return <pre><code className={className}>{children}</code></pre>;
+            }
+        }
     }
-  }
-});
+}
+
+Highlight.defaultProps = {
```
Comment (Calyhre) [inline src/index.js]: I'm curious, why not making it a static variable to the Highlight class?
Like this:
```javascript
class Highlight extends Component {
  static defaultProps = {
    // ...
  };
}
```

Comment (MichaelDeBoey) [inline src/index.js]: The recommanded way to use the `prop-types` package is like I did it and like you can see in the [docs](https://github.com/reactjs/prop-types#usage) :-)

**Comment by jerryshew** [comment]
@akiran when will you merge this PR?

**Comment by XaxisJosh** [comment]
Would love to see this merged in soon. it's causing some warnings in my app.

**Comment by maksim-lunochkin** [comment]
Merged this PR in forked version and updated it.
You can use this package instead:
https://www.npmjs.com/package/@maxlun86/react-highlight

**Comment by MichaelDeBoey** [comment]
Thanks for updating @akiran! 🙂

I think you can delete your scoped package @maksim-lunochkin 🙂

**Comment by maksim-lunochkin** [comment]
Yeah, deleted it

**Comment by akiran** [comment]
Thanks for the PR @MichaelDeBoey

**Comment by MichaelDeBoey** [comment]
Your welcome @akiran 🙂

**Pull request merged**